libs/api/ebpf_api.cpp

```diff
@@ -3334,15 +3335,45 @@ ebpf_object_load(_Inout_ struct bpf_object* object) NO_EXCEPT_TRY
     }
 
     if (Platform::_is_native_program(object->file_name)) {
+        // Currently _ebpf_program_load_native only supports one program type and attach type
+        // even if there are multiple programs in the file.  So look up the types and fail for
+        // now if there's more than one type to load.
+        ebpf_program_type_t program_type = EBPF_PROGRAM_TYPE_UNSPECIFIED;
+        ebpf_attach_type_t attach_type = EBPF_ATTACH_TYPE_UNSPECIFIED;
         for (auto& program : object->programs) {
             if (!program->autoload) {
                 continue;
             }
-            return _ebpf_program_load_native(
-                object->file_name, &program->program_type, &program->attach_type, object->execution_type, object);
+            if (program_type == EBPF_PROGRAM_TYPE_UNSPECIFIED) {
+                program_type = program->program_type;
+            } else if (program->program_type != program_type) {
+                EBPF_RETURN_RESULT(EBPF_INVALID_ARGUMENT);
```

Comment:
I think this check will regress the current behavior.

Even if bpf2c takes a single program type as input today, that is an optional argument, and the ELF file _can_ have programs with different program types (if they have the correct section names that can uniquely determine the program types). 

Also, I looked at the code again, and even though we pass a single program type to `_ebpf_program_load_native()` which eventually passes it to ebpfcore, that program type is (rightly) ignored while loading programs from native module.

So I think we can do the following:
1. We can remove this check for now (and pass the program type of the first program)
2. In subsequent PRs, update the IOCTL `EBPF_OPERATION_LOAD_NATIVE_PROGRAMS` to not take any program type as input.
3. In subsequent PRs, also update `bpf_program__set_type()` to fail the call for native modules.

Reply:
`bpf_program__set_type()` is now updated in this PR to fail the call for native modules.

Reply:
Also did items 1 and 2.